src/components/NcDialogButton/NcDialogButton.vue

```diff
@@ -99,16 +105,30 @@ export default defineComponent({
 	emits: ['click'],
 
 	setup(props, { emit }) {
+		const isLoading = ref(false)
+
 		/**
 		 * Handle clicking the button
 		 * @param {MouseEvent} e The click event
 		 */
-		const handleClick = (e) => {
-			props.callback?.()
-			emit('click', e)
+		const handleClick = async (e) => {
+			// Do not re-emit while loading
+			if (isLoading.value) {
+				return
+			}
+
+			isLoading.value = true
+			try {
+				const response = await props.callback?.()
+				if (response !== false) {
+					emit('click', e)
+				}
```

Comment:
nitpick: to not feel like this is always an HTTP request.
```suggestion
				const result = await props.callback?.()
				if (result !== false) {
					emit('click', e)
				}
```

Reply:
Maybe add it to the event payload?

```suggestion
				const response = await props.callback?.()
				if (response !== false) {
					emit('click', e, response)
				}
```

Reply:
For me it was not a HTTP response but the "response of the callback", but also fine with naming it result :sweat_smile: